src/test/strategy.test.ts

```diff
@@ -0,0 +1,40 @@
+import { SupabaseStrategy } from '../index'
+import { supabaseClient } from '../mocks/supabase'
+import { sessionStorage } from '../mocks/sessionStorage'
+
+describe('strategy', async() => {
+  it('should throw if missing supabaseClient', async() => {
+    expect.assertions(1)
+    // @ts-ignore
+    expect(() => new SupabaseStrategy()).toThrow('SupabaseStrategy : Constructor expected to receive a supabase client instance. Missing options.supabaseClient')
+  })
+  it('should throw if missing sessionStorage', async() => {
+    expect.assertions(1)
+    // @ts-ignore
+    expect(() => new SupabaseStrategy({ supabaseClient })).toThrow('SupabaseStrategy : Constructor expected to receive a session storage instance. Missing options.sessionStorage')
+  })
+  it('should throw if missing verify function', async() => {
+    expect.assertions(1)
+    // @ts-ignore
+    expect(() => new SupabaseStrategy({ supabaseClient, sessionStorage })).toThrow('SupabaseStrategy : Constructor expected to receive a verify function. Missing verify')
+  })
+  it('should throw if missing verify function', async() => {
+    expect.assertions(1)
+    // @ts-ignore
+    expect(() => new SupabaseStrategy({ supabaseClient, sessionStorage }, () => {})).not.toThrow()
+  })
+  it('should strategy provides default sessionKey and sessionErrorKey', async() => {
```

Comment:
I read this as; `it should ...` so maybe we can rephrase this to;

```suggestion
  it('should provide a default sessionKey and sessionErrorKey', async() => {
```